techmatters/hrm
# Chi 1991 parameterized queries

## Description

In order to make our data access logic safer from vulnerabilities, we should try to use PostgreSQL native parameterized queries, rather than relying on client side string interpolation provided as the default approach in pg-promise.

This PR attempts to make that easier by making using parameterized queries as straightforward as using client side interpolation. Out of the box, pg-promise mirrors the underlying postgres API, and only allows positional parameters. This PR adds an alternative factory function for pg-promise ParameterizedQuery objects which allow named parameters, as well as directly passing in JSON objects and arrays intended for use as a comma sewparated list (in an IN clause, for example).

It also converts the case data access, which contains some of our most complex data access logic, over to using parameterized queries as an example 

* Adds a helper function so allow the creation of PostgreSQL parameterized queries in pg-promise using named parameters.
* Converts the case data access module to use parameterized queries (the others would need doing later)

### Checklist

### Related Issues
CHI-1991

### Verification steps

Regression test case searching, lists and updating with this build of HRM deployed. You can also run a profiler and verify that the queries are hitting the database as parameters, not already interpolated into the SQL statement
